### PR TITLE
feat: move to Typescript code generation

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**

### CODE REVIEW

**Comment by xiaozhenliu-gg5** [comment]
Send out cl/299452868 for fixing the timeout setting in `admin/v2` which blocks system-test. Other than that, all tests are green.

File: synth.py [hunk 1]
```diff
@@ -5,46 +5,50 @@
...
+    "main-service": f"bigtable"
+    },
+    proto_path=f'/google/bigtable/{version}',
+    extra_proto_files=['google/cloud/common_resources.proto'],
+  )
 
 s.copy(
   v2_library,
-  excludes=['package.json', 'README.md', 'src/index.js', 'src/v2/index.js']
+  excludes=['package.json', 'README.md', 'src/index.ts', 'src/v2/index.ts', 'tsconfig.json', 'tslint.json']
```
Comment (JustinBeckwith) [inline synth.py]: If we truly need to ignore `index.ts`, `tsconfig`, etc, we need comment blocks explaining why please. 

Also, I don't want to ignore `tsconfig.json` and `tslint.json`

**Comment by steffnay** [comment]
@xiaozhenliu-gg5 @bcoe This looks good to me. I'm actually not super familiar with this library, though, so I don't think I can be much help with the tests

File: synth.py (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply synth.py]: `bigtable` is a little bit special, that it combines `admin/v2` and `v2` client files into `src/v2`. So there's manual work in `src/v2/index.ts` and `src/index.ts` to add AdminClients.

And for `tsconfig.json` & `tslint.json` it has enabled some setting like ` "noImplicitAny": false` to make sure the manual layer `src/*.ts` pass the lint check. 
I can surely remove these files from `excluded = []` and manually modify json files in synth.py, but that may increase many more lines in synth.py. Thank you!

Comment (JustinBeckwith) [reply synth.py]: Oh gross! 🙈  Skipping these for now is fine.  Can you please make sure there's a followup bug in the repo to fix the types and disable this though?  Maybe add a comment saying as much in this doc, referencing the issue?

Comment (xiaozhenliu-gg5) [reply synth.py]: yes, sure! I added some comments in synth.py, and I will also create bugs for them for better tracking. Thanks for the suggestions!

File: package.json [hunk 2]
```diff
@@ -69,38 +68,45 @@
     "through2": "^3.0.0"
   },
   "devDependencies": {
+    "@types/mocha": "^7.0.2",
```
Comment (alexander-fenster) [inline package.json]: nit: would be great to keep it sorted :)

**Comment by alexander-fenster** [comment]
@xiaozhenliu-gg5 Please remove `src/v2/.DS_Store ` from the change :)

File: synth.py [hunk 3]
```diff
@@ -5,46 +5,53 @@
...
 
 logging.basicConfig(level=logging.DEBUG)
-
-gapic = gcp.GAPICGenerator()
-v2_library = gapic.node_library(
+version='v2'
+gapic = gcp.GAPICMicrogenerator()
+v2_library = gapic.typescript_library(
   'bigtable',
   'v2',
```
Comment (alexander-fenster) [inline synth.py]: if you have `version='v2'`, let's use it here :)

Comment (xiaozhenliu-gg5) [reply synth.py]: sure!

File: package.json (hunk 2 above)
Comment (xiaozhenliu-gg5) [reply package.json]: of course, sorted it by `npx sort-package-json` :)

**Pull request merged**